src/scheduler.cc

```diff
@@ -35,27 +83,28 @@ __hidden ncclResult_t mscclSchedulerInit() {
   std::string mscclAlgoShareDirStr;
   const char *fullDirPath = nullptr;
   if (mscclAlgoDir == nullptr) {
-    // Try to find default algorithm directory based on librccl.so path
+    // Try to find default algorithm directory based on scheduler.so and shcheduler algo installtion path.
     Dl_info dl_info;
     struct link_map *link_map_ptr = nullptr;
-    if (!dladdr1((void *)ncclAllReduce, &dl_info, (void **)&link_map_ptr, RTLD_DL_LINKMAP)) {
+    if (!dladdr1((void *)mscclSchedulerInit, &dl_info, (void **)&link_map_ptr, RTLD_DL_LINKMAP)) {
       fprintf(stderr, "%s: dladdr1 failed", MSCCL_SCHEDULER_NAME);
       return ncclInvalidUsage;
     }
     std::string selfLibPath = link_map_ptr->l_name;
```

Comment:
Shall we keep aligned with https://github.com/ROCmSoftwarePlatform/rccl/blob/develop/src/misc/msccl/msccl_lifecycle.cc#L96-L103 , especially we should replace the second "find" to "find_last_of". This is a known bug that has been fixed in rccl.

Reply:
will keep the symbol named mscclSchedulerInit used for search for the scheduler.so, but already keep align the so file find path pattern.